runtime-decompiler/src/main/java/org/jrd/backend/data/ArchiveManager.java

```diff
@@ -0,0 +1,58 @@
+package org.jrd.backend.data;
+
+import org.jrd.backend.communication.FsAgent;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ArchiveManager {
+    File c;
+    String tmpdir = System.getProperty("java.ioo.tmpdir");
+    ArrayList<String> currentJars = new ArrayList<>();
+
+    public ArchiveManager(File c) {
+        this.c = c;
+    }
+
+    public boolean isClassInFile(String clazz) throws IOException {
+        ZipInputStream zis = new ZipInputStream(new FileInputStream(c));
+        if (!currentJars.isEmpty()) {
+            currentJars = new ArrayList<>();
+        }
+        currentJars.add(c.getName());
+        var ret = findClazz(zis, clazz);
+        if (!ret) {
+            currentJars.remove(c.getName());
+        }
+        return ret;
+    }
+
+    private boolean findClazz(ZipInputStream zis, String clazz) throws IOException {
+        ZipEntry entry = null;
+        while ((entry = zis.getNextEntry()) != null) {
+            if (!entry.isDirectory()) {
+                if (entry.getName().endsWith(".jar") || entry.getName().endsWith(".zip")) {
+                    currentJars.add(entry.getName());
+                    if(findClazz(new ZipInputStream(zis), clazz)) {
+                        return true;
+                    }
+                    currentJars.remove(entry.getName());
```

Comment:
I'm somehow failing why there are added strings to the list which are later removed. May you elaborate please?

Reply:
The Strings are added only in case the desired class was not found in this nested jar. thus later when we will be unpacking all the nested jars, only those that actually are required are unpacked. This results to depth first search through the file structure.

Reply:
whoo. Cool idea.
The only nit I can have is that it is severely unreadable from the code.
Maybe extract the list to separae class, and do the well-named operations here?

Reply:
Pls remove this to separate (prperly named) class with proeprly named methods